scylla/src/transport/load_balancing/default.rs

```diff
@@ -1487,12 +1491,13 @@ mod latency_awareness {
                 }),
                 Some(prev_avg) => Some({
                     let delay = (now - prev_avg.timestamp).as_secs_f64();
-                    let prev_weight = (delay + 1.).ln() / delay;
-                    let last_latency_nanos = last_latency.as_nanos() as f64;
-                    let prev_avg_nanos = prev_avg.average.as_nanos() as f64;
-                    let average = Duration::from_nanos(
-                        ((1. - prev_weight) * last_latency_nanos + prev_weight * prev_avg_nanos)
-                            .round() as u64,
+                    let scaled_delay = delay / scale_secs;
+                    let prev_weight = (delay + 1.).ln() / scaled_delay;
```

Comment:
Shouldn't this be `let prev_weight = (scaled_delay + 1.).ln() / scaled_delay`;?

Reply:
Indeed it should... thanks.